nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -55,6 +55,16 @@ in {
       '';
     };
 
+    configurationLimit = mkOption {
+      default = 0;
+      example = 120;
+      type = types.int;
+      description = ''
+        Maximum of configurations in boot menu. Otherwise boot partition could
+        run out of disk space.
```

Comment:
The meaning of the value 0 is not explained anywhere.

Reply:
Does this mean it will keep the newest 120 or oldest 120, or delete configurations, or garbage collect profiles, or?

Reply:
Pushed doc update

Reply:
Magic values like 0 should be avoided. Maybe give it type `nullOr int`?

Reply:
BTW the word "otherwise" implies that it *can't* run out of disk space if you do set a limit, which isn't the case.

Reply:
"Maximum of configurations" should be "Maximum number of configurations".

Reply:
I think I've addressed all comments.